File: app/src/main/java/com/bijoysingh/quicknote/views/ColorView.java
package com.bijoysingh.quicknote.views;

import android.annotation.TargetApi;
import android.content.Context;
import android.graphics.Color;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;

import com.bijoysingh.quicknote.R;
import com.bijoysingh.quicknote.utils.CircleDrawable;

public class ColorView extends LinearLayout {

  public View root;
  ImageView icon;

  public ColorView(Context context) {
    super(context);
    init(context);
  }

  public void init(Context context) {
    LayoutInflater inflater =
        (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    root = inflater.inflate(R.layout.layout_color, null);

    LayoutParams params =
        new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
    root.setLayoutParams(params);
    icon = (ImageView) root.findViewById(R.id.color_icon);
    addView(root);
  }

  public ColorView(Context context, AttributeSet attrs) {
    super(context, attrs);
    init(context);
  }

  public ColorView(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
    init(context);
  }

  @TargetApi(21)
  public ColorView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
    super(context, attrs, defStyleAttr, defStyleRes);
    init(context);
  }

  public void setColor(final int color, boolean selected) {
    this.icon.setImageResource(selected ? R.drawable.ic_done_white_48dp : 0);
    this.icon.setBackground(new CircleDrawable(color));
    this.icon.setColorFilter(Color.WHITE);
  }
}

File: base/src/main/java/com/maubis/scarlet/base/settings/view/ColorView.kt
package com.maubis.scarlet.base.settings.view

import android.annotation.TargetApi
import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout

import com.maubis.scarlet.base.R
import com.maubis.scarlet.base.support.ui.CircleDrawable

class ColorView : LinearLayout {

  lateinit internal var root: View
  lateinit internal var icon: ImageView

  constructor(context: Context) : super(context) {
    init(context)
  }


  constructor(context: Context, layout: Int) : super(context) {
    init(context, layout)
  }

  constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
    init(context)
  }

  constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
    init(context)
  }

  @TargetApi(21)
  constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
    init(context)
  }

  fun init(context: Context, layout: Int = R.layout.layout_color) {
    val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    root = inflater.inflate(layout, null)

    val params = LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
    root.layoutParams = params
    icon = root.findViewById<ImageView>(R.id.color_icon)
    addView(root)
  }

  fun setColor(color: Int, selected: Boolean) {
    this.icon.setImageResource(if (selected) R.drawable.ic_done_white_48dp else 0)
    this.icon.background = CircleDrawable(color)
    this.icon.setColorFilter(Color.WHITE)
  }
}
